projects/storefrontlib/cms-components/myaccount/my-coupons/claim-dialog/claim-dialog.component.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+  ElementRef,
+  HostListener,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { Subscription } from 'rxjs';
+import {
+  RoutingService,
+  CustomerCouponService,
+  GlobalMessageService,
+  GlobalMessageType,
+  useFeatureStyles,
+} from '@spartacus/core';
+import { FocusConfig, LaunchDialogService } from '../../../../layout/index';
+import { ICON_TYPE } from '../../../../cms-components/misc/icon/index';
+
+@Component({
+  selector: 'cx-claim-dialog',
+  templateUrl: './claim-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ClaimDialogComponent implements OnDestroy, OnInit {
+  private subscription = new Subscription();
+  iconTypes = ICON_TYPE;
+  private pageSize = 10;
+
+  couponCode: string;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
+
+  form: FormGroup = new FormGroup({
+    couponCode: new FormControl('', [Validators.required]),
+  });
+
+  @HostListener('click', ['$event'])
+  handleClick(event: UIEvent): void {
+    if ((event.target as any).tagName === this.el.nativeElement.tagName) {
+      this.close('Cross click');
+    }
+  }
+  constructor(
```

Comment:
`major:` For DI, let's use the `inject` function instead of a constructor to avoid introducing unnecessary breaking changes. See: https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines#CodingGuidelines-Useinject()functioninsteadofconstructorforDependencyInjection

Reply:
thanks, fixed.